### PR TITLE
Split the plugin-example into two binaries.

### LINKED ISSUE #11: more examples
Hi,

Would it be possible to see more implementation examples? I'm wrapping my head around on how to create dynamically loaded plugins ;/

### PR DESCRIPTION
**Pull request by skx**
This reflects actual usage, and will close #11.

### CODE REVIEW

File: examples/basic/main.go [hunk 1]
```diff
@@ -48,15 +41,6 @@ func main() {
...
-	// so that the host and our plugin can verify they can talk to each other.
-	// Then we set the plugin map to say what plugins we're serving.
-	plugin.Serve(&plugin.ServeConfig{
-		HandshakeConfig: handshakeConfig,
-		Plugins:         pluginMap,
-	})
-}
 
 // handshakeConfigs are used to just do a basic handshake between
 // a plugin and host. If the handshake fails, a user friendly error is shown.
```
Comment (xiaoyao1991) [inline examples/basic/main.go]: Are we supposed to remove the plugin implementations below?

File: examples/basic/main.go [hunk 2]
```diff
@@ -26,7 +19,7 @@ func main() {
 	client := plugin.NewClient(&plugin.ClientConfig{
 		HandshakeConfig: handshakeConfig,
 		Plugins:         pluginMap,
```
Comment (xiaoyao1991) [inline examples/basic/main.go]: I guess this is a more general usage question. Do we really need these configs, especially `pluginMap` here? That means it needs to know what plugins are available at compile time right? Kinda destroy the whole purpose of having plugins

Comment (Raggaer) [inline examples/basic/main.go]: I think the same. is this mandatory?

**Comment by mitchellh** [comment]
Whoops, I saw #20 first (top of list :( ). Sorry. Merged that which should fix the same issue! Thanks!

**Pull request closed without merging**